### PR TITLE
/api/articles/:article_id endpoint

### PR DESCRIPTION
**Pull request by jasper-howard**
ticket 4 complete with error tests for invalid and valid but none existent params

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -29,3 +29,60 @@ describe("3. GET /api/topics", () => {
...
+    return request(app).get("/api/articles/1").expect(200);
+  });
+  test("should respond with an article obj with correct properties", () => {
+    return request(app)
+      .get("/api/articles/2")
+      .then(({ body }) => {
+        const objOfCorrectShape = {
+          author: expect.any(String),
+          title: expect.any(String),
+          article_id: expect.any(Number),
```
Comment (karlaferry) [inline __tests__/api.test.js:44]: If you're making a request for a specific article_id, you want to make sure you're getting the right one back, so you should hardcode the id here :)

File: __tests__/api.test.js [hunk 2]
```diff
@@ -29,3 +29,60 @@ describe("3. GET /api/topics", () => {
...
+          body: expect.any(String),
+          topic: expect.any(String),
+          created_at: expect.any(String), // is there a timestamp costructor i need to use?
```
Comment (karlaferry) [inline __tests__/api.test.js:47]: Should be ok as a string for now!

File: __tests__/api.test.js [hunk 3]
```diff
@@ -29,3 +29,60 @@ describe("3. GET /api/topics", () => {
...
+          votes: expect.any(Number),
+        };
+
+        // console.log(body.article);
+        expect(body.article[0]).toMatchObject(objOfCorrectShape);
+      });
+  });
+  test("should respond with object specific to specified id", () => {
```
Comment (karlaferry) [inline __tests__/api.test.js:55]: You can def combine this and the previous test block if you want!

File: controllers/controllers.js [hunk 4]
```diff
@@ -1,7 +1,18 @@
-const { selectTopics } = require("../models/models");
+const { selectTopics, selectArticleById } = require("../models/models");
 
 exports.getTopics = (req, res, next) => {
   selectTopics().then((topics) => {
-    res.status(200).send(topics);
+    res.status(200).send(topics); // change to {topics: topic}??
```
Comment (karlaferry) [inline controllers/controllers.js:5]: Definitely send `{topics: topics}` or `{topics}`!

File: models/models.js [hunk 5]
```diff
@@ -5,3 +5,19 @@ exports.selectTopics = async () => {
...
+      `SELECT *FROM articles 
+             WHERE article_id = $1;`,
+      [id]
+    );
+    if (article.length === 0) {
+      return Promise.reject({ status: 404, msg: "not found" });
+    }
+    return article;
+  } catch (error) {
+    throw error;
```
Comment (karlaferry) [inline models/models.js:21]: Make sure you use `next(error)` instead of throw.

**Review by karlaferry — APPROVED** [review]
Looks good, just left some recommendations :)

File: models/models.js (hunk 5 above)
Comment (jasper-howard) [reply models/models.js:21]: I couldn't get next(error) to work, should it be a straight swap? Also thanks for answering my comment questions : )

**Pull request merged**